src/gaussian.cpp

```diff
@@ -257,6 +257,31 @@ Covar convolution(const Covar & C, const Covar & K)
     exp(-((x-m)%5E2*s%5E2+%2B+(y-n)%5E2*t%5E2+-+2*r*(x-m)*(y-n)*s*t)%2F(2*(1-r%5E2)))+wrt+s
 */
 
+struct ValuesGauss
+{
+    double cen_x;
+    double cen_y;
+    double L;
+    double sigma_x;
+    double sigma_y;
+    double rho;
+};
+
+typedef std::array<double, N_PARAMS> Weights;
+enum class OutputType : unsigned char
+{
+    none      = 0,
+    overwrite = 1,
+    add       = 2,
+    residual  = 3,
+};
+enum class GradientType : unsigned char
+{
+    none      = 0,
+    loglike   = 1,
+    jacobian  = 2,
+};
+
```

Comment:
This was a suggestion I had written down, but I'm glad to see it is now invalid.